src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -1534,6 +1564,209 @@ subroutine init_atm_static(mesh, dims, configs)
+
+    implicit none
+
+    ! Input variables
+    type (mpas_pool_type), intent(in) :: mesh
+    type (mpas_geotile_mgr_type), intent(in) :: mgr
+    type (mpas_kd_type), pointer, intent(in) :: kdtree
+    procedure (interp_criteria_function) :: interp_criteria
+    procedure (interp_accumulation_function) :: accumulation_method
+    integer, intent(in), optional :: supersample_fac
+
+    ! Local variables
+    integer, pointer :: nCells
+    integer, pointer :: nCellsSolve
+    integer, dimension(:), pointer :: bdyMaskCell
+    integer, dimension(:), pointer :: nEdgesOnCell
+    integer, dimension(:,:), pointer :: verticesOnCell
+    real (kind=RKIND), dimension(:), pointer :: xCell, yCell, zCell
+    real (kind=RKIND), dimension(:), pointer :: xVertex, yVertex, zVertex
+    real (kind=RKIND), pointer :: sphere_radius
+    real (kind=RKIND), dimension(:), pointer :: latCell, lonCell
+
+    integer, pointer :: tile_bdr
+    integer, pointer :: tile_nx, tile_ny
+    integer, pointer :: tile_z_start, tile_z_end
+    integer :: supersample_fac_lcl
+    integer :: subsample_fac
+
+    real (kind=RKIND) :: lat
+    real (kind=RKIND) :: lon
+    real (kind=RKIND) :: xPixel, yPixel, zPixel
+    real (kind=RKIND), pointer :: scale_factor
+
+    integer :: iCell
+    integer :: i, ii
+    integer :: j, jj
+
+    logical :: all_pixels_mapped_to_halo_cells
+
+    type (mpas_geotile_type), pointer :: tile
+    type (mpas_kd_type), pointer :: res
+
+    integer :: ierr
+
+    call mpas_pool_get_dimension(mesh, 'nCells', nCells)
+    call mpas_pool_get_dimension(mesh, 'nCellsSolve', nCellsSolve)
+    call mpas_pool_get_config(mesh, 'sphere_radius', sphere_radius)
+
+    call mpas_pool_get_array(mesh, 'latCell', latCell)
+    call mpas_pool_get_array(mesh, 'lonCell', lonCell)
+    call mpas_pool_get_array(mesh, 'bdyMaskCell', bdyMaskCell)
+    call mpas_pool_get_array(mesh, 'xCell', xCell)
+    call mpas_pool_get_array(mesh, 'yCell', yCell)
+    call mpas_pool_get_array(mesh, 'zCell', zCell)
+    call mpas_pool_get_array(mesh, 'xVertex', xVertex)
+    call mpas_pool_get_array(mesh, 'yVertex', yVertex)
+    call mpas_pool_get_array(mesh, 'zVertex', zVertex)
+    call mpas_pool_get_array(mesh, 'nEdgesOnCell', nEdgesOnCell)
+    call mpas_pool_get_array(mesh, 'verticesOnCell', verticesOnCell)
+    call mpas_pool_get_array(mesh, 'bdyMaskCell', bdyMaskCell)
+
+    call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+    call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+    call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+    call mpas_pool_get_config(mgr % pool, 'tile_z_start', tile_z_start)
+    call mpas_pool_get_config(mgr % pool, 'tile_z_end', tile_z_end)
+    call mpas_pool_get_config(mgr % pool, 'scale_factor', scale_factor)
+
+    ! Currently, the namelist option config_supersample_factor is only for max snow albedo and
+    ! monthly climatological albedo.
```

Comment:
Since this interpolation routine is generic, perhaps we don't need this comment that identifies specific fields, whose supersampling factor is the concern of the code that calls this routine.